<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>ScrambledTreeBuilder • ScrambledTreeBuilder</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ScrambledTreeBuilder">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ScrambledTreeBuilder</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/ScrambledTreeBuilder.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/brennern/ScrambledTreeBuilder/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>ScrambledTreeBuilder</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/brennern/ScrambledTreeBuilder/blob/master/vignettes/ScrambledTreeBuilder.Rmd" class="external-link"><code>vignettes/ScrambledTreeBuilder.Rmd</code></a></small>
      <div class="d-none name"><code>ScrambledTreeBuilder.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>cache <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_knit.html" class="external-link">opts_knit</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <em>ScrambledTreeBuilder</em> package consists of numerous data
formatting functions for phylogenetic tree building in the context of
the (still internal) <a href="https://github.com/oist/LuscombeU_ScramblingInTheTreeOfLife" class="external-link">Scrambling
in the Tree of Life</a> project.</p>
</div>
<div class="section level2">
<h2 id="load-package">Load Package<a class="anchor" aria-label="anchor" href="#load-package"></a>
</h2>
<p>The <em>ScrambledTreeBuilder</em> package outputs plots in
<em>ggplot2</em> format but you need to load the <em>ggplot2</em>
package to further customize them.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/brennern/ScrambledTreeBuilder" class="external-link">ScrambledTreeBuilder</a></span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>              <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example-data">Example Data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h2>
<p>This package utilizes example YAML files containing summary
statistics of halobacteria genome comparison data. In regards to genome
scrambling, many studies have showcased significant genome
rearrangements in such halobacteria species due to dozens of insertion
sequence families.</p>
<p>The YAML summary files are produced by performing an all vs. all
genome comparison between six halobacteria species using the <a href="https://nf-co.re/pairgenomealign/" class="external-link">nf-core pairwise genome
alignment pipeline</a> and an <a href="https://github.com/oist/plessy_nf_GenomicBreaks" class="external-link">post-processing
pipeline</a> running functions from the <a href="https://github.com/oist/GenomicBreaks" class="external-link">GenomicBreaks</a> package
to extract statistics on alignment length and the extent of genome
scrambling. Examples YAML files are stored in <code>extdata/yaml</code>.
Each file represent one pairwise alignment, and the file names conveys
the identifiers of the <em>target</em> and <em>query</em> genomes (here
species names) separed with three underscores (<code>___</code>).</p>
<p>Here we prepare an object called ‘yamlFileData’ that contains the
path to the files.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">yamlFileData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/yaml"</span>, package <span class="op">=</span> <span class="st">"ScrambledTreeBuilder"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/resultFiles.html">resultFiles</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">yamlFileData</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="co">#&gt;                                                                              Halobacterium_litoreum___Halobacterium_noricense </span></span>
<span><span class="co">#&gt; "/home/runner/work/_temp/Library/ScrambledTreeBuilder/extdata/yaml/Halobacterium_litoreum___Halobacterium_noricense.yaml.bz2"</span></span></code></pre></div>
<p>Next, we use the <code><a href="../reference/formatStats.html">formatStats()</a></code> function to load the YAML
files into a single dataframe where each line is a pair of species and
each column is a statistic or a metadata about that species
comparison.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exDataFrame</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/formatStats.html">formatStats</a></span><span class="op">(</span><span class="va">yamlFileData</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">exDataFrame</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 234</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">exDataFrame</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "aligned_length_Min"    "aligned_length_Q1"     "aligned_length_Median"</span></span>
<span><span class="co">#&gt; [4] "aligned_length_Mean"   "aligned_length_Q3"     "aligned_length_Max"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">exDataFrame</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span>pat <span class="op">=</span> <span class="st">"_Mean"</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "aligned_length_Mean"      "aligned_score_Mean"      </span></span>
<span><span class="co">#&gt; [3] "aligned_matches_Mean"     "aligned_mismatches_Mean" </span></span>
<span><span class="co">#&gt; [5] "aligned_gaps_target_Mean" "aligned_gaps_query_Mean"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">exDataFrame</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "percent_identity_global"   "percent_difference_local" </span></span>
<span><span class="co">#&gt; [3] "percent_difference_global" "index_avg_strandDiscord"  </span></span>
<span><span class="co">#&gt; [5] "percent_aligned"           "lab"</span></span></code></pre></div>
<p>This data frame has a large number of columns, providing summary
statistics on various aspects of the alignments. For statistics of
interest, we build square matrices where rows and columns indicate one
species, and the cells at each intersection contain the value for that
pair.</p>
<p>We perform this task with the <code><a href="../reference/makeMatrix.html">makeMatrix()</a></code> function. It
provide defaults for missing values and self-comparisons. In this
vignette, let’s focus on the percent nucleotide difference and the
scrambling index.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Percent nucleotide difference  We will use it to cluster a tree.</span></span>
<span><span class="va">treeMatrix</span> <span class="op">&lt;-</span> <span class="fl">100</span> <span class="op">-</span> <span class="fu"><a href="../reference/makeMatrix.html">makeMatrix</a></span><span class="op">(</span><span class="va">exDataFrame</span>, <span class="st">"percent_identity_global"</span>, <span class="fl">100</span>, <span class="fl">50</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">treeMatrix</span><span class="op">)</span></span>
<span><span class="co">#&gt;                         Halobacterium_litoreum Halobacterium_noricense</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                       0                      21</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                     21                       0</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                     24                      25</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei                      32                      34</span></span>
<span><span class="co">#&gt; Haloferax_volcanii                          30                      30</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                       27                      28</span></span>
<span><span class="co">#&gt;                         Halobacterium_salinarum Haloferax_mediterranei</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                       24                     31</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                      25                     32</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                       0                     33</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei                       33                      0</span></span>
<span><span class="co">#&gt; Haloferax_volcanii                           32                     20</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                        28                     31</span></span>
<span><span class="co">#&gt;                         Haloferax_volcanii Salarchaeum_japonicum</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                  30                    27</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                 30                    28</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                 32                    28</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei                  20                    32</span></span>
<span><span class="co">#&gt; Haloferax_volcanii                       0                    30</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                   30                     0</span></span>
<span><span class="co">#&gt; attr(,"builtWith")</span></span>
<span><span class="co">#&gt; [1] "percent_identity_global"</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Scrambling index</span></span>
<span><span class="va">valueMatrix</span> <span class="op">&lt;-</span> <span class="fl">1</span> <span class="op">-</span> <span class="fu"><a href="../reference/makeMatrix.html">makeMatrix</a></span><span class="op">(</span><span class="va">exDataFrame</span>, <span class="st">"index_avg_strandRand"</span>, <span class="fl">1</span>, <span class="fl">0.5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">valueMatrix</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#&gt;                         Halobacterium_litoreum Halobacterium_noricense</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                    0.00                    0.61</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                   0.61                    0.00</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                   0.47                    0.77</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei                    0.92                    0.93</span></span>
<span><span class="co">#&gt; Haloferax_volcanii                        0.94                    0.95</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                     0.82                    0.87</span></span>
<span><span class="co">#&gt;                         Halobacterium_salinarum Haloferax_mediterranei</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                     0.47                   0.92</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                    0.77                   0.93</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                    0.00                   0.93</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei                     0.93                   0.00</span></span>
<span><span class="co">#&gt; Haloferax_volcanii                         0.87                   0.68</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                      0.92                   0.89</span></span>
<span><span class="co">#&gt;                         Haloferax_volcanii Salarchaeum_japonicum</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                0.94                  0.81</span></span>
<span><span class="co">#&gt; Halobacterium_noricense               0.95                  0.87</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum               0.88                  0.92</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei                0.68                  0.89</span></span>
<span><span class="co">#&gt; Haloferax_volcanii                    0.00                  0.99</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                 0.99                  0.00</span></span>
<span><span class="co">#&gt; attr(,"builtWith")</span></span>
<span><span class="co">#&gt; [1] "index_avg_strandRand"</span></span></code></pre></div>
<p>We cluster the percent nucleotide difference matrix
(<code>treeMatrix</code>) to produce a tree in <em>tibble</em> format,
using the <code><a href="../reference/makeTidyTree.html">makeTidyTree()</a></code> function.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Let's average the target-query and query-target replicate pairs.</span></span>
<span><span class="op">(</span><span class="va">Tibble</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeTidyTree.html">makeTidyTree</a></span><span class="op">(</span><span class="op">(</span><span class="va">treeMatrix</span><span class="op">/</span><span class="fl">2</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">treeMatrix</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: `aes_()` was deprecated in ggplot2 3.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use tidy evaluation idioms with `aes()`</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">ggtree</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/YuLab-SMU/ggtree/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Warning in fortify(data, ...): Arguments in `...` must be used.</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> Problematic arguments:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> as.Date = as.Date</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> yscale_mapping = yscale_mapping</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> hang = hang</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you misspell an argument name?</span></span>
<span><span class="co">#&gt; Warning: `aes_string()` was deprecated in ggplot2 3.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use tidy evaluation idioms with `aes()`.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> See also `vignette("ggplot2-in-packages")` for more information.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">ggtree</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/YuLab-SMU/ggtree/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tbl_tree abstraction: 11 × 6</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># which can be converted to treedata or phylo </span></span></span>
<span><span class="co"><span style="color: #949494;">#&gt; # via as.treedata or as.phylo</span></span></span>
<span><span class="co">#&gt;    parent  node branch.length label                   isTip     y</span></span>
<span><span class="co">#&gt;     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>     11     1         10.6  Halobacterium_litoreum  TRUE   5   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>     11     2         10.6  Halobacterium_noricense TRUE   6   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>     10     3         12.3  Halobacterium_salinarum TRUE   4   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>      8     4          9.85 Haloferax_mediterranei  TRUE   1   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>      8     5          9.85 Haloferax_volcanii      TRUE   2   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>      9     6         13.9  Salarchaeum_japonicum   TRUE   3   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>      7     7         <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>                      FALSE  2.69</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>      7     8          5.79 <span style="color: #BB0000;">NA</span>                      FALSE  1.5 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>      7     9          1.75 <span style="color: #BB0000;">NA</span>                      FALSE  3.88</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>      9    10          1.57 <span style="color: #BB0000;">NA</span>                      FALSE  4.75</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span>     10    11          1.70 <span style="color: #BB0000;">NA</span>                      FALSE  5.5</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualizeTree.html">visualizeTree</a></span><span class="op">(</span><span class="va">Tibble</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in fortify(data, ...): Arguments in `...` must be used.</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> Problematic arguments:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> as.Date = as.Date</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> yscale_mapping = yscale_mapping</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> hang = hang</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you misspell an argument name?</span></span></code></pre></div>
<p><img src="ScrambledTreeBuilder_files/figure-html/Tree_with_node_IDs-1.png" width="700"></p>
<p>The node IDs can be used to manipulate the tree, for instance
subsetting with the <code>subtree()</code> function.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualizeTree.html">visualizeTree</a></span><span class="op">(</span><span class="va">Tibble</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/subTree.html">subTree</a></span><span class="op">(</span><span class="fl">9</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in fortify(data, ...): Arguments in `...` must be used.</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> Problematic arguments:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> as.Date = as.Date</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> yscale_mapping = yscale_mapping</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> hang = hang</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you misspell an argument name?</span></span></code></pre></div>
<p><img src="ScrambledTreeBuilder_files/figure-html/subTree_with_node_IDs-1.png" width="700"></p>
<p>Clades of interest can be tracked and highlighted by
<code>FocalClade</code> objects. For robustness against changes in the
input data or clustering approach, it is recommented to define a clade
by the most recent common ancestor from a pair of species. The
<code><a href="../reference/subTree.html">subTree()</a></code> function can take <code>FocalClade</code> objects
instead of node IDs as input.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">Halobacterium</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/focalClade.html">focalClade</a></span><span class="op">(</span><span class="va">Tibble</span>, <span class="st">"Halobacterium_noricense"</span>, <span class="st">"Halobacterium_salinarum"</span>, <span class="st">"blue"</span>, <span class="st">"Halobacterium genus"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Halobacterium genus, node ID: 10, number of genomes: 3</span></span>
<span><span class="va">Halobacterium</span><span class="op">@</span><span class="va">genomeIDs</span></span>
<span><span class="co">#&gt; [1] "Halobacterium_litoreum"  "Halobacterium_noricense"</span></span>
<span><span class="co">#&gt; [3] "Halobacterium_salinarum"</span></span>
<span><span class="va">Halobacterium</span><span class="op">@</span><span class="va">nodeID</span></span>
<span><span class="co">#&gt; [1] 10</span></span>
<span><span class="fu"><a href="../reference/subTree.html">subTree</a></span><span class="op">(</span><span class="va">Tibble</span>, <span class="va">Halobacterium</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tbl_tree abstraction: 5 × 6</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># which can be converted to treedata or phylo </span></span></span>
<span><span class="co"><span style="color: #949494;">#&gt; # via as.treedata or as.phylo</span></span></span>
<span><span class="co">#&gt;   parent  node branch.length label                   group node.orig</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>      5     1         10.6  Halobacterium_litoreum  1             1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span>      5     2         10.6  Halobacterium_noricense 1             2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span>      4     3         12.3  Halobacterium_salinarum 1             3</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span>      4     4          3.32 <span style="color: #BB0000;">NA</span>                      1            10</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span>      4     5          1.70 <span style="color: #BB0000;">NA</span>                      1            11</span></span></code></pre></div>
<p>The focal clade objects can be added to plots to highlight the clades
in the selected colors.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Haloferax</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/focalClade.html">focalClade</a></span><span class="op">(</span><span class="va">Tibble</span>, <span class="st">"Haloferax_mediterranei"</span>, <span class="st">"Haloferax_volcanii"</span>, <span class="st">"green3"</span>, <span class="st">"Haloferax genus"</span><span class="op">)</span></span>
<span><span class="op">(</span><span class="va">clades</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FocalCladeList.html">FocalCladeList</a></span><span class="op">(</span><span class="va">Halobacterium</span>, <span class="va">Haloferax</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Halobacterium genus, node ID: 10, number of genomes: 3</span></span>
<span><span class="co">#&gt; Haloferax genus, node ID: 8, number of genomes: 2</span></span>
<span><span class="fu"><a href="../reference/visualizeTree.html">visualizeTree</a></span><span class="op">(</span><span class="va">Tibble</span><span class="op">)</span> <span class="op">+</span> <span class="va">clades</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Focal clade highlight"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in fortify(data, ...): Arguments in `...` must be used.</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> Problematic arguments:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> as.Date = as.Date</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> yscale_mapping = yscale_mapping</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> hang = hang</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you misspell an argument name?</span></span></code></pre></div>
<p><img src="ScrambledTreeBuilder_files/figure-html/focalCladeList-1.png" width="700"></p>
<p>To plot more data on the tree, we add other statistics to the tree
object, here the scrambling index and the percent nucleotide difference,
using the <code><a href="../reference/makeValueTibble.html">makeValueTibble()</a></code> function. This operation
reduces a pairwise matrix to the tree, by averaging all the pairs
sharing the same most recent common ancestor, represented by an internal
node in the tree.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">tibbleWithValues</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeValueTibble.html">makeValueTibble</a></span><span class="op">(</span><span class="va">Tibble</span>, <span class="va">valueMatrix</span>, colname <span class="op">=</span> <span class="st">"Scrambling_index"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tbl_tree abstraction: 11 × 7</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># which can be converted to treedata or phylo </span></span></span>
<span><span class="co"><span style="color: #949494;">#&gt; # via as.treedata or as.phylo</span></span></span>
<span><span class="co">#&gt;    parent  node branch.length label                 isTip     y Scrambling_index</span></span>
<span><span class="co">#&gt;     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>     11     1         10.6  Halobacterium_litore… TRUE   5              <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>     11     2         10.6  Halobacterium_norice… TRUE   6              <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>     10     3         12.3  Halobacterium_salina… TRUE   4              <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>      8     4          9.85 Haloferax_mediterran… TRUE   1              <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>      8     5          9.85 Haloferax_volcanii    TRUE   2              <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>      9     6         13.9  Salarchaeum_japonicum TRUE   3              <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>      7     7         <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>                    FALSE  2.69            0.926</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>      7     8          5.79 <span style="color: #BB0000;">NA</span>                    FALSE  1.5             0.683</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>      7     9          1.75 <span style="color: #BB0000;">NA</span>                    FALSE  3.88            0.869</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>      9    10          1.57 <span style="color: #BB0000;">NA</span>                    FALSE  4.75            0.618</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span>     10    11          1.70 <span style="color: #BB0000;">NA</span>                    FALSE  5.5             0.605</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">tibbleWithMultipleValues</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeValueTibble.html">makeValueTibble</a></span><span class="op">(</span><span class="va">tibbleWithValues</span>, <span class="va">treeMatrix</span>, colname <span class="op">=</span> <span class="st">"Percent_difference"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tbl_tree abstraction: 11 × 8</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># which can be converted to treedata or phylo </span></span></span>
<span><span class="co"><span style="color: #949494;">#&gt; # via as.treedata or as.phylo</span></span></span>
<span><span class="co">#&gt;    parent  node branch.length label                 isTip     y Scrambling_index</span></span>
<span><span class="co">#&gt;     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>     11     1         10.6  Halobacterium_litore… TRUE   5              <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>     11     2         10.6  Halobacterium_norice… TRUE   6              <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>     10     3         12.3  Halobacterium_salina… TRUE   4              <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>      8     4          9.85 Haloferax_mediterran… TRUE   1              <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>      8     5          9.85 Haloferax_volcanii    TRUE   2              <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>      9     6         13.9  Salarchaeum_japonicum TRUE   3              <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>      7     7         <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>                    FALSE  2.69            0.926</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>      7     8          5.79 <span style="color: #BB0000;">NA</span>                    FALSE  1.5             0.683</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>      7     9          1.75 <span style="color: #BB0000;">NA</span>                    FALSE  3.88            0.869</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>      9    10          1.57 <span style="color: #BB0000;">NA</span>                    FALSE  4.75            0.618</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span>     10    11          1.70 <span style="color: #BB0000;">NA</span>                    FALSE  5.5             0.605</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: Percent_difference &lt;dbl&gt;</span></span></span></code></pre></div>
<p>We made multiple tables to show the step-by-step process, but
typically one would just keep the last table. This can be done with
pipes.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/makeTidyTree.html">makeTidyTree</a></span><span class="op">(</span><span class="op">(</span><span class="va">treeMatrix</span><span class="op">/</span><span class="fl">2</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">treeMatrix</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/makeValueTibble.html">makeValueTibble</a></span><span class="op">(</span><span class="va">valueMatrix</span>, colname <span class="op">=</span> <span class="st">"Scrambling_index"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/makeValueTibble.html">makeValueTibble</a></span><span class="op">(</span><span class="va">treeMatrix</span>,  colname <span class="op">=</span> <span class="st">"Percent_difference"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in fortify(data, ...): Arguments in `...` must be used.</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> Problematic arguments:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> as.Date = as.Date</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> yscale_mapping = yscale_mapping</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> hang = hang</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you misspell an argument name?</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tbl_tree abstraction: 11 × 8</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># which can be converted to treedata or phylo </span></span></span>
<span><span class="co"><span style="color: #949494;">#&gt; # via as.treedata or as.phylo</span></span></span>
<span><span class="co">#&gt;    parent  node branch.length label                 isTip     y Scrambling_index</span></span>
<span><span class="co">#&gt;     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>     11     1         10.6  Halobacterium_litore… TRUE   5              <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>     11     2         10.6  Halobacterium_norice… TRUE   6              <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>     10     3         12.3  Halobacterium_salina… TRUE   4              <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>      8     4          9.85 Haloferax_mediterran… TRUE   1              <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>      8     5          9.85 Haloferax_volcanii    TRUE   2              <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>      9     6         13.9  Salarchaeum_japonicum TRUE   3              <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>      7     7         <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>                    FALSE  2.69            0.926</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>      7     8          5.79 <span style="color: #BB0000;">NA</span>                    FALSE  1.5             0.683</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>      7     9          1.75 <span style="color: #BB0000;">NA</span>                    FALSE  3.88            0.869</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>      9    10          1.57 <span style="color: #BB0000;">NA</span>                    FALSE  4.75            0.618</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span>     10    11          1.70 <span style="color: #BB0000;">NA</span>                    FALSE  5.5             0.605</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: Percent_difference &lt;dbl&gt;</span></span></span>
<span><span class="co"># Same result</span></span></code></pre></div>
<div class="section level3">
<h3 id="plot-trees-with-values">Plot trees with values<a class="anchor" aria-label="anchor" href="#plot-trees-with-values"></a>
</h3>
<p>Let’s use the <code>tibbleWithMultipleValues</code> object to plot
trees. In our case, we have generated a tree built based on nucleotide
percent difference values as a distance, and computed average scrambling
index for all the nodes. We can plot these values as labels on the
tree.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualizeTree.html">visualizeTree</a></span><span class="op">(</span><span class="va">tibbleWithMultipleValues</span>, <span class="st">"Scrambling_index"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Tree built with Percent difference and labelled with Scrambling Index"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="va">clades</span></span>
<span><span class="co">#&gt; Warning in fortify(data, ...): Arguments in `...` must be used.</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> Problematic arguments:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> as.Date = as.Date</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> yscale_mapping = yscale_mapping</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> hang = hang</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you misspell an argument name?</span></span></code></pre></div>
<p><img src="ScrambledTreeBuilder_files/figure-html/tree_with_SI-1.png" width="700"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualizeTree.html">visualizeTree</a></span><span class="op">(</span><span class="va">tibbleWithMultipleValues</span>, <span class="va">tibbleWithMultipleValues</span><span class="op">$</span><span class="va">Scrambling_index</span>, ynudge <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Tree labeled with Scrambling Index and Percent Difference"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_color_viridis_c</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Scrambling index"</span>, option <span class="op">=</span> <span class="st">"cividis"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggnewscale</span><span class="fu">::</span><span class="fu"><a href="https://eliocamp.github.io/ggnewscale/reference/new_scale.html" class="external-link">new_scale_colour</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtree</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_label</a></span><span class="op">(</span><span class="fu">ggtree</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">Percent_difference</span><span class="op">)</span>, color <span class="op">=</span> <span class="va">Percent_difference</span><span class="op">)</span>, label.size <span class="op">=</span> <span class="fl">0.25</span>, size <span class="op">=</span> <span class="fl">3</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span>, label.padding <span class="op">=</span> <span class="fu">ggtree</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">0.15</span>, <span class="st">"lines"</span><span class="op">)</span>, nudge_y <span class="op">=</span> <span class="op">-</span><span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridis/reference/scale_viridis.html" class="external-link">scale_color_viridis</a></span><span class="op">(</span>option <span class="op">=</span> <span class="st">"magma"</span>, name <span class="op">=</span> <span class="st">"Percent Identity"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in fortify(data, ...): Arguments in `...` must be used.</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> Problematic arguments:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> as.Date = as.Date</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> yscale_mapping = yscale_mapping</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> hang = hang</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you misspell an argument name?</span></span>
<span><span class="co">#&gt; Warning: The `label.size` argument of `geom_label()` is deprecated as of ggplot2 3.5.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use the `linewidth` argument instead.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span></code></pre></div>
<p><img src="ScrambledTreeBuilder_files/figure-html/two_value_plot-1.png" width="700"></p>
<p>Of course, if you spotted an interesting sub-tree, you can plot the
node IDs to easily extract it for further analysis.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualizeTree.html">visualizeTree</a></span><span class="op">(</span><span class="va">tibbleWithMultipleValues</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in fortify(data, ...): Arguments in `...` must be used.</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> Problematic arguments:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> as.Date = as.Date</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> yscale_mapping = yscale_mapping</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> hang = hang</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you misspell an argument name?</span></span></code></pre></div>
<p><img src="ScrambledTreeBuilder_files/figure-html/tree_with_node_IDs_again-1.png" width="700"></p>
<p>The <code>subTree</code> function can conveniently be used with R’s
pipe operator to cut a sub-tree at a chosen node.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualizeTree.html">visualizeTree</a></span><span class="op">(</span><span class="va">tibbleWithMultipleValues</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/subTree.html">subTree</a></span><span class="op">(</span>node <span class="op">=</span> <span class="fl">9</span><span class="op">)</span>, <span class="st">"Percent_difference"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in fortify(data, ...): Arguments in `...` must be used.</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> Problematic arguments:</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> as.Date = as.Date</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> yscale_mapping = yscale_mapping</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> hang = hang</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Did you misspell an argument name?</span></span></code></pre></div>
<p><img src="ScrambledTreeBuilder_files/figure-html/subTree_node_9-1.png" width="700"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/subMatrix.html">subMatrix</a></span><span class="op">(</span><span class="va">Tibble</span>, <span class="va">valueMatrix</span>, <span class="fl">9</span>, simpl<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt;             H_litoreum H_noricense H_salinarum S_japonicum</span></span>
<span><span class="co">#&gt; H_litoreum   0.0000000   0.6054199   0.4662469   0.8119173</span></span>
<span><span class="co">#&gt; H_noricense  0.6073435   0.0000000   0.7667464   0.8669942</span></span>
<span><span class="co">#&gt; H_salinarum  0.4677172   0.7676303   0.0000000   0.9210253</span></span>
<span><span class="co">#&gt; S_japonicum  0.8192124   0.8669109   0.9218979   0.0000000</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plot-node-and-leaf-values">Plot node and leaf values<a class="anchor" aria-label="anchor" href="#plot-node-and-leaf-values"></a>
</h3>
<p>We can also plot leaf and node values in a standard scatterplot.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fl">100</span> <span class="op">-</span> <span class="va">exDataFrame</span><span class="op">$</span><span class="va">percent_identity_global</span>, <span class="fl">1</span> <span class="op">-</span> <span class="va">exDataFrame</span><span class="op">$</span><span class="va">index_avg_strandRand</span>, col <span class="op">=</span> <span class="st">'grey'</span>, xlab <span class="op">=</span> <span class="st">"Percent difference"</span>, ylab<span class="op">=</span><span class="st">"Strand randomisation index"</span>, main <span class="op">=</span> <span class="st">"Leaf values are in grey and node values in red"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="va">tibbleWithMultipleValues</span><span class="op">$</span><span class="va">Percent_difference</span>, <span class="va">tibbleWithMultipleValues</span><span class="op">$</span><span class="va">Scrambling_index</span>, col <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">plotTwoBranches</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tree</span>, <span class="va">node</span>, <span class="va">X</span>, <span class="va">Y</span>, <span class="va">...</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">children</span> <span class="op">&lt;-</span> <span class="fu">tidytree</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/tidytree/man/child.html" class="external-link">child</a></span>   <span class="op">(</span><span class="va">tree</span>, <span class="va">node</span><span class="op">)</span></span>
<span>  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">children</span><span class="op">)</span> <span class="op">!=</span> <span class="fl">2</span><span class="op">)</span> <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">parent</span>   <span class="op">&lt;-</span> <span class="fu">tidytree</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/tidytree/man/parent.html" class="external-link">parent</a></span><span class="op">(</span><span class="va">tree</span>, <span class="va">children</span><span class="op">$</span><span class="va">node</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">parent</span><span class="op">[</span>, <span class="va">X</span>, drop<span class="op">=</span><span class="cn">T</span><span class="op">]</span>, <span class="va">children</span><span class="op">[</span>, <span class="va">X</span>, drop<span class="op">=</span><span class="cn">T</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">parent</span><span class="op">[</span>, <span class="va">Y</span>, drop<span class="op">=</span><span class="cn">T</span><span class="op">]</span>, <span class="va">children</span><span class="op">[</span>, <span class="va">Y</span>, drop<span class="op">=</span><span class="cn">T</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>,</span>
<span>    <span class="va">...</span></span>
<span>  <span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">parent</span><span class="op">[</span>, <span class="va">X</span>, drop<span class="op">=</span><span class="cn">T</span><span class="op">]</span>, <span class="va">children</span><span class="op">[</span>, <span class="va">X</span>, drop<span class="op">=</span><span class="cn">T</span><span class="op">]</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">parent</span><span class="op">[</span>, <span class="va">Y</span>, drop<span class="op">=</span><span class="cn">T</span><span class="op">]</span>, <span class="va">children</span><span class="op">[</span>, <span class="va">Y</span>, drop<span class="op">=</span><span class="cn">T</span><span class="op">]</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>,</span>
<span>    <span class="va">...</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">plotAllBranches</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tree</span>, <span class="va">X</span>, <span class="va">Y</span>, <span class="va">...</span> <span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">tree</span><span class="op">$</span><span class="va">parent</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span>\<span class="op">(</span><span class="va">node</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu">plotTwoBranches</span><span class="op">(</span><span class="va">tree</span>, <span class="va">node</span>, <span class="va">X</span>, <span class="va">Y</span>, <span class="va">...</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">plotAllBranches</span><span class="op">(</span><span class="va">tibbleWithMultipleValues</span>, <span class="st">"Percent_difference"</span>, <span class="st">"Scrambling_index"</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></code></pre></div>
<p><img src="ScrambledTreeBuilder_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Noa Brenner, Charles Plessy, Anika Mittal.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
