<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>ScrambledTreeBuilder • ScrambledTreeBuilder</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ScrambledTreeBuilder">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ScrambledTreeBuilder</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9001</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/ScrambledTreeBuilder.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/brennern/ScrambledTreeBuilder/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>ScrambledTreeBuilder</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/brennern/ScrambledTreeBuilder/blob/master/vignettes/ScrambledTreeBuilder.Rmd" class="external-link"><code>vignettes/ScrambledTreeBuilder.Rmd</code></a></small>
      <div class="d-none name"><code>ScrambledTreeBuilder.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>cache <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_knit.html" class="external-link">opts_knit</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <em>ScrambledTreeBuilder</em> package consists of numerous data
formatting functions for phylogenetic tree building in the context of
the (still internal) <a href="https://github.com/oist/LuscombeU_ScramblingInTheTreeOfLife" class="external-link">Scrambling
in the Tree of Life</a> project.</p>
</div>
<div class="section level2">
<h2 id="load-package">Load Package<a class="anchor" aria-label="anchor" href="#load-package"></a>
</h2>
<p>The <em>ScrambledTreeBuilder</em> package ouptuts plots in
<em>ggplot2</em> format but you need to load the <em>ggplot2</em>
package to further customize them.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/brennern/ScrambledTreeBuilder" class="external-link">ScrambledTreeBuilder</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example-data">Example Data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h2>
<p>This package utilizes example <code>.yaml</code> files containing
summary statistics of halobacteria genome comparison data. In regards to
genome scrambling, many studies have showcased significant genome
rearrangments in such halobacteria species due to dozens of insertion
sequence families.</p>
<p>The YAML summary files are produced by performing an all vs. all
genome comparison between six halobacteria species using the Nextflow <a href="https://github.com/oist/plessy_pairwiseGenomeComparison" class="external-link">genome
pairwise alignment pipeline</a> and an <a href="https://github.com/brennern/GenomeScrambling/blob/main/scripts/makeGBreaksInputFile.sh" class="external-link">input
file script</a> for <a href="https://github.com/oist/GenomicBreaks" class="external-link">GenomicBreaks</a>. They are
stored in a directory called ‘extdata/PairwiseComparisons’. Here we
prepare an object called ‘yamlFileData’ that contains the path to the
files. The file names are constructed by taking the name of one species
taken as reference, adding a separator (<code>___</code>), and then the
name of one species being compared to the reference.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">resultsDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/PairwiseComparisons"</span>, package <span class="op">=</span> <span class="st">"ScrambledTreeBuilder"</span><span class="op">)</span></span>
<span><span class="va">yamlFileData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">resultsDir</span>, pattern <span class="op">=</span> <span class="st">"*.yaml.bz2"</span>, full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">yamlFileData</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">yamlFileData</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/basename.html" class="external-link">basename</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span>pat <span class="op">=</span> <span class="st">".yaml.bz2"</span>, rep<span class="op">=</span><span class="st">""</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">yamlFileData</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Halobacterium_litoreum___Halobacterium_noricense"          </span></span>
<span><span class="co">#&gt; [2] "Halobacterium_litoreum___Halobacterium_salinarum"          </span></span>
<span><span class="co">#&gt; [3] "Halobacterium_litoreum___Haloferax_mediterranei_ATCC_33500"</span></span>
<span><span class="co">#&gt; [4] "Halobacterium_litoreum___Haloferax_volcanii_DS2"           </span></span>
<span><span class="co">#&gt; [5] "Halobacterium_litoreum___Salarchaeum_japonicum"            </span></span>
<span><span class="co">#&gt; [6] "Halobacterium_noricense___Halobacterium_litoreum"</span></span></code></pre></div>
<p>Next, we use the ‘formatStats()’ function to load the YAML files into
a single dataframe where each line is a pair of species and each column
is a statistic or a metadata about that species comparison.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exDataFrame</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/formatStats.html">formatStats</a></span><span class="op">(</span><span class="va">yamlFileData</span><span class="op">)</span></span></code></pre></div>
<p>Then, for statistics of interest, we build square matrices where rows
and columns indicate one species, and the cells at each intersection
contain the value for that pair.</p>
<p>We perform this task with the ‘makeMatrix()’ function. It provide
defaults for missing values and self-comparisons.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Average nucleotide identity.  We will use it to cluster a tree.</span></span>
<span><span class="op">(</span><span class="va">treeMatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeMatrix.html">makeMatrix</a></span><span class="op">(</span><span class="va">exDataFrame</span>, <span class="st">"percent_identity_global"</span>, <span class="fl">100</span>, <span class="fl">50</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                   Halobacterium_noricense</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                         100.00000</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                          74.94759</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500                66.42841</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                           69.93183</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                            72.11068</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                           78.81199</span></span>
<span><span class="co">#&gt;                                   Halobacterium_salinarum</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                          74.94623</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                         100.00000</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500                66.97889</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                           68.48311</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                            71.80855</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                           75.75208</span></span>
<span><span class="co">#&gt;                                   Haloferax_mediterranei_ATCC_33500</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                                    68.45333</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                                    67.00552</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500                         100.00000</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                                     80.39441</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                                      68.65929</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                                     68.72077</span></span>
<span><span class="co">#&gt;                                   Haloferax_volcanii_DS2 Salarchaeum_japonicum</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                         69.96147              72.42201</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                         68.48878              71.55815</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500               80.22559              68.41933</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                         100.00000              69.79642</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                           69.82446             100.00000</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                          70.16293              72.74950</span></span>
<span><span class="co">#&gt;                                   Halobacterium_litoreum</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                         78.73936</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                         75.84298</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500               68.27405</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                          70.11358</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                           72.75352</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                         100.00000</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Strand randomisation index.</span></span>
<span><span class="op">(</span><span class="va">valueMatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeMatrix.html">makeMatrix</a></span><span class="op">(</span><span class="va">exDataFrame</span>, <span class="st">"index_avg_strandRand"</span>, <span class="fl">1</span>, <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                   Halobacterium_noricense</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                        1.00000000</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                        0.23236975</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500              0.07235525</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                         0.05324080</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                          0.13308910</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                         0.39458005</span></span>
<span><span class="co">#&gt;                                   Halobacterium_salinarum</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                        0.23325365</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                        1.00000000</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500              0.07435565</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                         0.12659775</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                          0.07810210</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                         0.53375310</span></span>
<span><span class="co">#&gt;                                   Haloferax_mediterranei_ATCC_33500</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                                  0.07203045</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                                  0.07326215</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500                        1.00000000</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                                   0.31779245</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                                    0.10685560</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                                   0.07664500</span></span>
<span><span class="co">#&gt;                                   Haloferax_volcanii_DS2 Salarchaeum_japonicum</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                       0.05398890            0.13300580</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                       0.12422790            0.07897470</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500             0.31721145            0.10753295</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                        1.00000000            0.01033135</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                         0.00958875            1.00000000</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                        0.06412845            0.18808275</span></span>
<span><span class="co">#&gt;                                   Halobacterium_litoreum</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                       0.39265650</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                       0.53228275</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500             0.08363770</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                        0.06379615</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                         0.18078755</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                        1.00000000</span></span></code></pre></div>
<p>We cluster the average nucleotide identity matrix
(<code>treeMatrix</code>) to produce a tree in <em>tibble</em> format,
using the <code><a href="../reference/makeTidyTree.html">makeTidyTree()</a></code> funciton. Then we add other
statistics, here the strand randomisation index and the average
nucleotide identity, to this object using the
<code>makeValueTibble</code> function.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Let's average the target-query and query-target replicate pairs.</span></span>
<span><span class="op">(</span><span class="va">Tibble</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeTidyTree.html">makeTidyTree</a></span><span class="op">(</span><span class="fl">100</span> <span class="op">-</span> <span class="op">(</span><span class="va">treeMatrix</span><span class="op">/</span><span class="fl">2</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">treeMatrix</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tbl_tree abstraction: 11 × 4</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># which can be converted to treedata or phylo </span></span></span>
<span><span class="co"><span style="color: #949494;">#&gt; # via as.treedata or as.phylo</span></span></span>
<span><span class="co">#&gt;    parent  node branch.length label                            </span></span>
<span><span class="co">#&gt;     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>     11     1         10.6  Halobacterium_noricense          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>     10     2         12.3  Halobacterium_salinarum          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>      8     3          9.85 Haloferax_mediterranei_ATCC_33500</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>      8     4          9.85 Haloferax_volcanii_DS2           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>      9     5         13.9  Salarchaeum_japonicum            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>     11     6         10.6  Halobacterium_litoreum           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>      7     7         <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>                               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>      7     8          5.79 <span style="color: #BB0000;">NA</span>                               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>      7     9          1.75 <span style="color: #BB0000;">NA</span>                               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>      9    10          1.57 <span style="color: #BB0000;">NA</span>                               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span>     10    11          1.70 <span style="color: #BB0000;">NA</span></span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">tibbleWithValue</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeValueTibble.html">makeValueTibble</a></span><span class="op">(</span><span class="va">Tibble</span>, <span class="va">valueMatrix</span>, colname <span class="op">=</span> <span class="st">"Strand_Randomisation_Index"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tbl_tree abstraction: 11 × 5</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># which can be converted to treedata or phylo </span></span></span>
<span><span class="co"><span style="color: #949494;">#&gt; # via as.treedata or as.phylo</span></span></span>
<span><span class="co">#&gt;    parent  node branch.length label                       Strand_Randomisation…¹</span></span>
<span><span class="co">#&gt;     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>     11     1         10.6  Halobacterium_noricense                    <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>     10     2         12.3  Halobacterium_salinarum                    <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>      8     3          9.85 Haloferax_mediterranei_ATC…                <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>      8     4          9.85 Haloferax_volcanii_DS2                     <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>      9     5         13.9  Salarchaeum_japonicum                      <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>     11     6         10.6  Halobacterium_litoreum                     <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>      7     7         <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>                                          0.074<span style="text-decoration: underline;">0</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>      7     8          5.79 <span style="color: #BB0000;">NA</span>                                          0.317 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>      7     9          1.75 <span style="color: #BB0000;">NA</span>                                          0.131 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>      9    10          1.57 <span style="color: #BB0000;">NA</span>                                          0.382 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span>     10    11          1.70 <span style="color: #BB0000;">NA</span>                                          0.393 </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ abbreviated name: ¹​Strand_Randomisation_Index</span></span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">tibbleWithMultipleValues</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeValueTibble.html">makeValueTibble</a></span><span class="op">(</span><span class="va">tibbleWithValue</span>, <span class="va">treeMatrix</span>, colname <span class="op">=</span> <span class="st">"Percent_Identity"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tbl_tree abstraction: 11 × 6</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># which can be converted to treedata or phylo </span></span></span>
<span><span class="co"><span style="color: #949494;">#&gt; # via as.treedata or as.phylo</span></span></span>
<span><span class="co">#&gt;    parent  node branch.length label      Strand_Randomisation…¹ Percent_Identity</span></span>
<span><span class="co">#&gt;     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>     11     1         10.6  Halobacte…                <span style="color: #BB0000;">NA</span>                  <span style="color: #BB0000;">NA</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>     10     2         12.3  Halobacte…                <span style="color: #BB0000;">NA</span>                  <span style="color: #BB0000;">NA</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>      8     3          9.85 Haloferax…                <span style="color: #BB0000;">NA</span>                  <span style="color: #BB0000;">NA</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>      8     4          9.85 Haloferax…                <span style="color: #BB0000;">NA</span>                  <span style="color: #BB0000;">NA</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>      9     5         13.9  Salarchae…                <span style="color: #BB0000;">NA</span>                  <span style="color: #BB0000;">NA</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>     11     6         10.6  Halobacte…                <span style="color: #BB0000;">NA</span>                  <span style="color: #BB0000;">NA</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>      7     7         <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>                         0.074<span style="text-decoration: underline;">0</span>             68.6</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>      7     8          5.79 <span style="color: #BB0000;">NA</span>                         0.317              80.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>      7     9          1.75 <span style="color: #BB0000;">NA</span>                         0.131              72.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>      9    10          1.57 <span style="color: #BB0000;">NA</span>                         0.382              75.4</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span>     10    11          1.70 <span style="color: #BB0000;">NA</span>                         0.393              78.7</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ abbreviated name: ¹​Strand_Randomisation_Index</span></span></span></code></pre></div>
<p>We made multiple tables to show the step-by-step process, but
typically one would just keep the last table. This can be done with
pipes.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/makeTidyTree.html">makeTidyTree</a></span><span class="op">(</span><span class="fl">100</span> <span class="op">-</span> <span class="op">(</span><span class="va">treeMatrix</span><span class="op">/</span><span class="fl">2</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">treeMatrix</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/makeValueTibble.html">makeValueTibble</a></span><span class="op">(</span><span class="va">valueMatrix</span>, colname <span class="op">=</span> <span class="st">"Strand_Randomisation_Index"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/makeValueTibble.html">makeValueTibble</a></span><span class="op">(</span><span class="va">treeMatrix</span>, colname <span class="op">=</span> <span class="st">"Percent_Identity"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tbl_tree abstraction: 11 × 6</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># which can be converted to treedata or phylo </span></span></span>
<span><span class="co"><span style="color: #949494;">#&gt; # via as.treedata or as.phylo</span></span></span>
<span><span class="co">#&gt;    parent  node branch.length label      Strand_Randomisation…¹ Percent_Identity</span></span>
<span><span class="co">#&gt;     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>     11     1         10.6  Halobacte…                <span style="color: #BB0000;">NA</span>                  <span style="color: #BB0000;">NA</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>     10     2         12.3  Halobacte…                <span style="color: #BB0000;">NA</span>                  <span style="color: #BB0000;">NA</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>      8     3          9.85 Haloferax…                <span style="color: #BB0000;">NA</span>                  <span style="color: #BB0000;">NA</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>      8     4          9.85 Haloferax…                <span style="color: #BB0000;">NA</span>                  <span style="color: #BB0000;">NA</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>      9     5         13.9  Salarchae…                <span style="color: #BB0000;">NA</span>                  <span style="color: #BB0000;">NA</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>     11     6         10.6  Halobacte…                <span style="color: #BB0000;">NA</span>                  <span style="color: #BB0000;">NA</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>      7     7         <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>                         0.074<span style="text-decoration: underline;">0</span>             68.6</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>      7     8          5.79 <span style="color: #BB0000;">NA</span>                         0.317              80.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>      7     9          1.75 <span style="color: #BB0000;">NA</span>                         0.131              72.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>      9    10          1.57 <span style="color: #BB0000;">NA</span>                         0.382              75.4</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span>     10    11          1.70 <span style="color: #BB0000;">NA</span>                         0.393              78.7</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ abbreviated name: ¹​Strand_Randomisation_Index</span></span></span>
<span><span class="co"># Same result</span></span></code></pre></div>
<div class="section level3">
<h3 id="plot-trees">Plot trees<a class="anchor" aria-label="anchor" href="#plot-trees"></a>
</h3>
<p>Let’s use the <code>tibbleWithMultipleValues</code> object to plot
trees. In our case, we have generated a tree built based on 100 minus
percent identity values as a distance, and computed average strand
randomisation scores for all the nodes. We can plot these values as
labels on the tree.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualizeTree.html">visualizeTree</a></span><span class="op">(</span><span class="va">tibbleWithMultipleValues</span>, <span class="va">tibbleWithMultipleValues</span><span class="op">$</span><span class="va">Strand_Randomisation_Index</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Tree built with Percent Identity and labelled with Strand Randomisation Index scores"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggtree</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom-hilight.html" class="external-link">geom_hilight</a></span><span class="op">(</span>node <span class="op">=</span> <span class="fl">8</span>, fill <span class="op">=</span> <span class="st">"lightblue1"</span>, alpha <span class="op">=</span> <span class="fl">.4</span>, type <span class="op">=</span> <span class="st">"gradient"</span>, gradient.direction <span class="op">=</span> <span class="st">'tr'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtree</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom-hilight.html" class="external-link">geom_hilight</a></span><span class="op">(</span>node <span class="op">=</span> <span class="fl">9</span>, fill <span class="op">=</span> <span class="st">"pink"</span>, alpha <span class="op">=</span> <span class="fl">.4</span>, type <span class="op">=</span> <span class="st">"gradient"</span>, gradient.direction <span class="op">=</span> <span class="st">'tr'</span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">y</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">y</span>, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="ScrambledTreeBuilder_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualizeTree.html">visualizeTree</a></span><span class="op">(</span><span class="va">tibbleWithMultipleValues</span>, <span class="va">tibbleWithMultipleValues</span><span class="op">$</span><span class="va">Strand_Randomisation_Index</span>, ynudge <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Tree labeled with Strand Randomisation Index and Percent Identity (built with Percent Identity)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_color_viridis_c</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Strand Randomisation Index"</span>, option <span class="op">=</span> <span class="st">"cividis"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggnewscale</span><span class="fu">::</span><span class="fu"><a href="https://eliocamp.github.io/ggnewscale/reference/new_scale.html" class="external-link">new_scale_colour</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtree</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_label</a></span><span class="op">(</span><span class="fu">ggtree</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">Percent_Identity</span>, digits <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>, color <span class="op">=</span> <span class="va">Percent_Identity</span><span class="op">)</span>, label.size <span class="op">=</span> <span class="fl">0.25</span>, size <span class="op">=</span> <span class="fl">3</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span>, label.padding <span class="op">=</span> <span class="fu">ggtree</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">0.15</span>, <span class="st">"lines"</span><span class="op">)</span>, nudge_y <span class="op">=</span> <span class="op">-</span><span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridis/reference/scale_viridis.html" class="external-link">scale_color_viridis</a></span><span class="op">(</span>option <span class="op">=</span> <span class="st">"magma"</span>, name <span class="op">=</span> <span class="st">"Percent Identity"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">y</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">y</span>, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="ScrambledTreeBuilder_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>Of course, if you spotted an interesting sub-tree, you can plot the
node IDs to easily extract it for further analysis.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualizeTree.html">visualizeTree</a></span><span class="op">(</span><span class="va">tibbleWithMultipleValues</span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">y</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">y</span>, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="ScrambledTreeBuilder_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<p>The <code>subTree</code> function can conveniently be used with R’s
pipe operator to cut a sub-tree at a chosen node.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualizeTree.html">visualizeTree</a></span><span class="op">(</span><span class="va">tibbleWithMultipleValues</span> <span class="op">|&gt;</span> <span class="fu"><a href="../reference/subTree.html">subTree</a></span><span class="op">(</span>node <span class="op">=</span> <span class="fl">9</span><span class="op">)</span>, <span class="st">"Percent_Identity"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">y</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">y</span>, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="ScrambledTreeBuilder_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/subMatrix.html">subMatrix</a></span><span class="op">(</span><span class="va">Tibble</span>, <span class="va">valueMatrix</span>, <span class="fl">9</span>, simpl<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt;             H_noricense H_salinarum S_japonicum H_litoreum</span></span>
<span><span class="co">#&gt; H_noricense   1.0000000   0.2332537   0.1330058  0.3926565</span></span>
<span><span class="co">#&gt; H_salinarum   0.2323697   1.0000000   0.0789747  0.5322828</span></span>
<span><span class="co">#&gt; S_japonicum   0.1330891   0.0781021   1.0000000  0.1807875</span></span>
<span><span class="co">#&gt; H_litoreum    0.3945801   0.5337531   0.1880827  1.0000000</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plot-node-and-leaf-values">Plot node and leaf values<a class="anchor" aria-label="anchor" href="#plot-node-and-leaf-values"></a>
</h3>
<p>We can also plot leaf and node values in a standard scatterplot.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">exDataFrame</span><span class="op">$</span><span class="va">percent_identity_global</span>, <span class="va">exDataFrame</span><span class="op">$</span><span class="va">index_avg_strandRand</span>, col <span class="op">=</span> <span class="st">'grey'</span>, xlab <span class="op">=</span> <span class="st">"Percent identity"</span>, ylab<span class="op">=</span><span class="st">"Strand randomisation index"</span>, main <span class="op">=</span> <span class="st">"Leaf values are in grey and node values in red"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="va">tibbleWithMultipleValues</span><span class="op">$</span><span class="va">Percent_Identity</span>, <span class="va">tibbleWithMultipleValues</span><span class="op">$</span><span class="va">Strand_Randomisation_Index</span>, col <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">plotTwoBranches</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tree</span>, <span class="va">node</span>, <span class="va">X</span>, <span class="va">Y</span>, <span class="va">...</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">children</span> <span class="op">&lt;-</span> <span class="fu">tidytree</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/tidytree/man/child.html" class="external-link">child</a></span>   <span class="op">(</span><span class="va">tree</span>, <span class="va">node</span><span class="op">)</span></span>
<span>  <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">children</span><span class="op">)</span> <span class="op">!=</span> <span class="fl">2</span><span class="op">)</span> <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">parent</span>   <span class="op">&lt;-</span> <span class="fu">tidytree</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/tidytree/man/parent.html" class="external-link">parent</a></span><span class="op">(</span><span class="va">tree</span>, <span class="va">children</span><span class="op">$</span><span class="va">node</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">parent</span><span class="op">[</span>, <span class="va">X</span>, drop<span class="op">=</span><span class="cn">T</span><span class="op">]</span>, <span class="va">children</span><span class="op">[</span>, <span class="va">X</span>, drop<span class="op">=</span><span class="cn">T</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">parent</span><span class="op">[</span>, <span class="va">Y</span>, drop<span class="op">=</span><span class="cn">T</span><span class="op">]</span>, <span class="va">children</span><span class="op">[</span>, <span class="va">Y</span>, drop<span class="op">=</span><span class="cn">T</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>,</span>
<span>    <span class="va">...</span></span>
<span>  <span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">parent</span><span class="op">[</span>, <span class="va">X</span>, drop<span class="op">=</span><span class="cn">T</span><span class="op">]</span>, <span class="va">children</span><span class="op">[</span>, <span class="va">X</span>, drop<span class="op">=</span><span class="cn">T</span><span class="op">]</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">parent</span><span class="op">[</span>, <span class="va">Y</span>, drop<span class="op">=</span><span class="cn">T</span><span class="op">]</span>, <span class="va">children</span><span class="op">[</span>, <span class="va">Y</span>, drop<span class="op">=</span><span class="cn">T</span><span class="op">]</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>,</span>
<span>    <span class="va">...</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">plotAllBranches</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tree</span>, <span class="va">X</span>, <span class="va">Y</span>, <span class="va">...</span> <span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">tree</span><span class="op">$</span><span class="va">parent</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span>\<span class="op">(</span><span class="va">node</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu">plotTwoBranches</span><span class="op">(</span><span class="va">tree</span>, <span class="va">node</span>, <span class="va">X</span>, <span class="va">Y</span>, <span class="va">...</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">plotAllBranches</span><span class="op">(</span><span class="va">tibbleWithMultipleValues</span>, <span class="st">"Percent_Identity"</span>, <span class="st">"Strand_Randomisation_Index"</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></code></pre></div>
<p><img src="ScrambledTreeBuilder_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Noa Brenner.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
