<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="ScrambledTreeBuilder">
<title>ScrambledTreeBuilder • ScrambledTreeBuilder</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ScrambledTreeBuilder">
<meta property="og:description" content="ScrambledTreeBuilder">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">ScrambledTreeBuilder</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/ScrambledTreeBuilder.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/brennern/ScrambledTreeBuilder/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>ScrambledTreeBuilder</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/brennern/ScrambledTreeBuilder/blob/HEAD/vignettes/ScrambledTreeBuilder.Rmd" class="external-link"><code>vignettes/ScrambledTreeBuilder.Rmd</code></a></small>
      <div class="d-none name"><code>ScrambledTreeBuilder.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>cache <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_knit.html" class="external-link">opts_knit</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The ScrambledTreeBuilder package consists of numerous data formatting
functions for phylogenetic tree building.</p>
</div>
<div class="section level2">
<h2 id="load-package">Load Package<a class="anchor" aria-label="anchor" href="#load-package"></a>
</h2>
<p>Whilst the ScrambledTreeBuilder package contains all necessary
functions required to convert data from .yaml files to a data type
suitable for phylogenetic trees, loading ggplot2 will allow for greater
customization of the final tree products.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/brennern/ScrambledTreeBuilder" class="external-link">ScrambledTreeBuilder</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example-data">Example Data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h2>
<p>This package utilizes example .yaml files from halobacteria genome
comparison data. In regards to genome scrambling, many studies have
showcased significant genome rearrangments in such halobacteria species
due to dozens of insertion sequence families. Thus, the
ScrambledTreeBuilder package will be useful in uncovering relationships
amongst such species.</p>
<p>Upon performing an all vs. all genome comparison between six
halobacteria species using the Nextflow <a href="https://github.com/oist/plessy_pairwiseGenomeComparison" class="external-link">genome
pairwise alignment pipeline</a> and an <a href="https://github.com/brennern/GenomeScrambling/blob/main/scripts/makeGBreaksInputFile.sh" class="external-link">input
file script</a> for <a href="https://github.com/oist/GenomicBreaks" class="external-link">GenomicBreaks</a>, we have
multiple .yaml files stored in a directory called
‘extdata/PairwiseComparisons’. We will load this data into RStudio with
the variable name ‘yamlFileData’.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">resultsDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/PairwiseComparisons"</span>, package <span class="op">=</span> <span class="st">"ScrambledTreeBuilder"</span><span class="op">)</span></span>
<span><span class="va">yamlFileData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">resultsDir</span>, pattern <span class="op">=</span> <span class="st">"*.yaml.bz2"</span>, full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">yamlFileData</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">yamlFileData</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/basename.html" class="external-link">basename</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span>pat <span class="op">=</span> <span class="st">".yaml.bz2"</span>, rep<span class="op">=</span><span class="st">""</span><span class="op">)</span></span></code></pre></div>
<p>Next, we can use the ‘formatStats()’ function to transform the .yaml
files into a single dataframe. Having a dataframe containing our species
information is important, as this will allow us to extract specific
columns/variables and plot them onto our phylogenetic trees later.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exDataFrame</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/formatStats.html">formatStats</a></span><span class="op">(</span><span class="va">yamlFileData</span><span class="op">)</span></span></code></pre></div>
<p>We will need to further process our data by transforming the
dataframe into a matrix. We can perform this task with the
‘makeMatrix()’ function. This function can be quite useful, as you may
use it to both create the matrix which you wish to build your tree with
and to create the matrix to store species information for a particular
variable. For example, we may want to build the tree based on percent
identity values, which we store in ‘treeMatrix’. Then, we may want to
plot strand randomisation index scores on the trees to study genome
scrambling, which we store in ‘valueMatrix’.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">valuesToBuildTheTree</span> <span class="op">&lt;-</span> <span class="st">"percent_identity_global"</span></span>
<span><span class="va">treeMatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeMatrix.html">makeMatrix</a></span><span class="op">(</span><span class="va">exDataFrame</span>, <span class="va">valuesToBuildTheTree</span>, <span class="fl">100</span>, <span class="fl">50</span><span class="op">)</span></span>
<span><span class="va">treeMatrix</span></span>
<span><span class="co">#&gt;                                   Halobacterium_noricense</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                         100.00000</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                          74.74168</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500                66.51318</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                           69.96063</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                            71.94129</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                           78.87231</span></span>
<span><span class="co">#&gt;                                   Halobacterium_salinarum</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                          74.72164</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                         100.00000</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500                67.06152</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                           68.47755</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                            71.82141</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                           75.78684</span></span>
<span><span class="co">#&gt;                                   Haloferax_mediterranei_ATCC_33500</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                                    68.55420</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                                    67.06788</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500                         100.00000</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                                     80.98460</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                                      68.70265</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                                     68.88111</span></span>
<span><span class="co">#&gt;                                   Haloferax_volcanii_DS2 Salarchaeum_japonicum</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                         69.98252              72.22600</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                         68.48654              71.57797</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500               80.80996              68.50482</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                         100.00000              69.84616</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                           69.87968             100.00000</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                          70.25591              72.76736</span></span>
<span><span class="co">#&gt;                                   Halobacterium_litoreum</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                         78.80471</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                         75.89491</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500               68.49688</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                          70.22052</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                           72.77139</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                         100.00000</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">valuesToPlaceOnLabels</span> <span class="op">&lt;-</span> <span class="st">"index_avg_strandRand"</span></span>
<span><span class="va">valueMatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeMatrix.html">makeMatrix</a></span><span class="op">(</span><span class="va">exDataFrame</span>, <span class="va">valuesToPlaceOnLabels</span>, <span class="fl">1</span>, <span class="fl">0.5</span><span class="op">)</span></span>
<span><span class="va">valueMatrix</span></span>
<span><span class="co">#&gt;                                   Halobacterium_noricense</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                        1.00000000</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                        0.23871495</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500              0.04880815</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                         0.02784920</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                          0.11476675</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                         0.39885460</span></span>
<span><span class="co">#&gt;                                   Halobacterium_salinarum</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                        0.23860920</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                        1.00000000</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500              0.07503800</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                         0.13530080</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                          0.06630175</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                         0.53733160</span></span>
<span><span class="co">#&gt;                                   Haloferax_mediterranei_ATCC_33500</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                                  0.04765400</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                                  0.07367205</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500                        1.00000000</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                                   0.33807060</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                                    0.10473430</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                                   0.07850880</span></span>
<span><span class="co">#&gt;                                   Haloferax_volcanii_DS2 Salarchaeum_japonicum</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                       0.02749270             0.1146204</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                       0.13299345             0.0671487</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500             0.33675900             0.1063006</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                        1.00000000             0.0044876</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                         0.00848495             1.0000000</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                        0.04474345             0.1884153</span></span>
<span><span class="co">#&gt;                                   Halobacterium_litoreum</span></span>
<span><span class="co">#&gt; Halobacterium_noricense                       0.39714395</span></span>
<span><span class="co">#&gt; Halobacterium_salinarum                       0.53626875</span></span>
<span><span class="co">#&gt; Haloferax_mediterranei_ATCC_33500             0.08857840</span></span>
<span><span class="co">#&gt; Haloferax_volcanii_DS2                        0.04606195</span></span>
<span><span class="co">#&gt; Salarchaeum_japonicum                         0.18109200</span></span>
<span><span class="co">#&gt; Halobacterium_litoreum                        1.00000000</span></span></code></pre></div>
<p>Using the ‘treeMatrix’, we will create an hclust object and use this
to produce a tibble dataset. Transforming into a tibble dataset is
necessary as this will allow us to add the variable from our
‘valueMatrix’ onto the tree with correct label placement on each node
without changing the structure of the tree. Once we have the tibble
dataset, we can utilize the ‘makeValueTibble()’ function to add the
value from our ‘valueMatrix’ to the tibble data.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">HClust</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html" class="external-link">hclust</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">dist</a></span><span class="op">(</span><span class="va">treeMatrix</span><span class="op">)</span>, method <span class="op">=</span> <span class="st">"complete"</span><span class="op">)</span></span>
<span><span class="va">Tibble</span> <span class="op">&lt;-</span> <span class="fu">tidytree</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="fu">tidytree</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/as.phylo.html" class="external-link">as.phylo</a></span><span class="op">(</span><span class="va">HClust</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">Tibble</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tbl_tree abstraction: 11 × 4</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># which can be converted to treedata or phylo </span></span></span>
<span><span class="co"><span style="color: #949494;">#&gt; # via as.treedata or as.phylo</span></span></span>
<span><span class="co">#&gt;    parent  node branch.length label                            </span></span>
<span><span class="co">#&gt;     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>     11     1         15.0  Halobacterium_noricense          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>     10     2         18.0  Halobacterium_salinarum          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>      8     3         13.7  Haloferax_mediterranei_ATCC_33500</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>      8     4         13.7  Haloferax_volcanii_DS2           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>      9     5         20.1  Salarchaeum_japonicum            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>     11     6         15.0  Halobacterium_litoreum           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>      7     7         <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>                               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>      7     8         11.1  <span style="color: #BB0000;">NA</span>                               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>      7     9          4.62 <span style="color: #BB0000;">NA</span>                               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>      9    10          2.19 <span style="color: #BB0000;">NA</span>                               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span>     10    11          2.98 <span style="color: #BB0000;">NA</span></span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tibbleWithValue</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeValueTibble.html">makeValueTibble</a></span><span class="op">(</span><span class="va">Tibble</span>, <span class="va">valueMatrix</span>, colname <span class="op">=</span> <span class="st">"Strand_Randomisation_Index"</span><span class="op">)</span></span>
<span><span class="va">tibbleWithValue</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tbl_tree abstraction: 11 × 5</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># which can be converted to treedata or phylo </span></span></span>
<span><span class="co"><span style="color: #949494;">#&gt; # via as.treedata or as.phylo</span></span></span>
<span><span class="co">#&gt;    parent  node branch.length label                       Strand_Randomisation…¹</span></span>
<span><span class="co">#&gt;     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>     11     1         15.0  Halobacterium_noricense                    <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>     10     2         18.0  Halobacterium_salinarum                    <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>      8     3         13.7  Haloferax_mediterranei_ATC…                <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>      8     4         13.7  Haloferax_volcanii_DS2                     <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>      9     5         20.1  Salarchaeum_japonicum                      <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>     11     6         15.0  Halobacterium_litoreum                     <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>      7     7         <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>                                          0.066<span style="text-decoration: underline;">6</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>      7     8         11.1  <span style="color: #BB0000;">NA</span>                                          0.337 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>      7     9          4.62 <span style="color: #BB0000;">NA</span>                                          0.121 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>      9    10          2.19 <span style="color: #BB0000;">NA</span>                                          0.387 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span>     10    11          2.98 <span style="color: #BB0000;">NA</span>                                          0.397 </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ abbreviated name: ¹​Strand_Randomisation_Index</span></span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tibbleWithMultipleValues</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeValueTibble.html">makeValueTibble</a></span><span class="op">(</span><span class="va">tibbleWithValue</span>, <span class="va">treeMatrix</span>, colname <span class="op">=</span> <span class="st">"Percent_Identity"</span><span class="op">)</span></span>
<span><span class="va">tibbleWithMultipleValues</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tbl_tree abstraction: 11 × 6</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># which can be converted to treedata or phylo </span></span></span>
<span><span class="co"><span style="color: #949494;">#&gt; # via as.treedata or as.phylo</span></span></span>
<span><span class="co">#&gt;    parent  node branch.length label      Strand_Randomisation…¹ Percent_Identity</span></span>
<span><span class="co">#&gt;     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>     11     1         15.0  Halobacte…                <span style="color: #BB0000;">NA</span>                  <span style="color: #BB0000;">NA</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>     10     2         18.0  Halobacte…                <span style="color: #BB0000;">NA</span>                  <span style="color: #BB0000;">NA</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>      8     3         13.7  Haloferax…                <span style="color: #BB0000;">NA</span>                  <span style="color: #BB0000;">NA</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>      8     4         13.7  Haloferax…                <span style="color: #BB0000;">NA</span>                  <span style="color: #BB0000;">NA</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>      9     5         20.1  Salarchae…                <span style="color: #BB0000;">NA</span>                  <span style="color: #BB0000;">NA</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>     11     6         15.0  Halobacte…                <span style="color: #BB0000;">NA</span>                  <span style="color: #BB0000;">NA</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>      7     7         <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>                         0.066<span style="text-decoration: underline;">6</span>             68.6</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>      7     8         11.1  <span style="color: #BB0000;">NA</span>                         0.337              80.8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>      7     9          4.62 <span style="color: #BB0000;">NA</span>                         0.121              72.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>      9    10          2.19 <span style="color: #BB0000;">NA</span>                         0.387              75.3</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span>     10    11          2.98 <span style="color: #BB0000;">NA</span>                         0.397              78.8</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ abbreviated name: ¹​Strand_Randomisation_Index</span></span></span></code></pre></div>
<p>Finally, we can use the ‘tibbleWithValue’ and
‘tibbleWithMultipleValues’ datasets to build the tree. In our case, we
will generate a tree built based on percent identity values with strand
randomisation scores labeled on all nodes using ‘tibbleWithValue’.
‘tibbleWithMultipleValues’ will allow us to plot both strand
randomisation index scores and percent identity scores on the tree.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">SingleValueTree</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/visualizeTree.html">visualizeTree</a></span><span class="op">(</span><span class="va">tibbleWithValue</span>, <span class="va">tibbleWithValue</span><span class="op">$</span><span class="va">Strand_Randomisation_Index</span><span class="op">)</span></span>
<span><span class="co">#&gt; Registered S3 methods overwritten by 'treeio':</span></span>
<span><span class="co">#&gt;   method              from    </span></span>
<span><span class="co">#&gt;   MRCA.phylo          tidytree</span></span>
<span><span class="co">#&gt;   MRCA.treedata       tidytree</span></span>
<span><span class="co">#&gt;   Nnode.treedata      tidytree</span></span>
<span><span class="co">#&gt;   Ntip.treedata       tidytree</span></span>
<span><span class="co">#&gt;   ancestor.phylo      tidytree</span></span>
<span><span class="co">#&gt;   ancestor.treedata   tidytree</span></span>
<span><span class="co">#&gt;   child.phylo         tidytree</span></span>
<span><span class="co">#&gt;   child.treedata      tidytree</span></span>
<span><span class="co">#&gt;   full_join.phylo     tidytree</span></span>
<span><span class="co">#&gt;   full_join.treedata  tidytree</span></span>
<span><span class="co">#&gt;   groupClade.phylo    tidytree</span></span>
<span><span class="co">#&gt;   groupClade.treedata tidytree</span></span>
<span><span class="co">#&gt;   groupOTU.phylo      tidytree</span></span>
<span><span class="co">#&gt;   groupOTU.treedata   tidytree</span></span>
<span><span class="co">#&gt;   inner_join.phylo    tidytree</span></span>
<span><span class="co">#&gt;   inner_join.treedata tidytree</span></span>
<span><span class="co">#&gt;   is.rooted.treedata  tidytree</span></span>
<span><span class="co">#&gt;   nodeid.phylo        tidytree</span></span>
<span><span class="co">#&gt;   nodeid.treedata     tidytree</span></span>
<span><span class="co">#&gt;   nodelab.phylo       tidytree</span></span>
<span><span class="co">#&gt;   nodelab.treedata    tidytree</span></span>
<span><span class="co">#&gt;   offspring.phylo     tidytree</span></span>
<span><span class="co">#&gt;   offspring.treedata  tidytree</span></span>
<span><span class="co">#&gt;   parent.phylo        tidytree</span></span>
<span><span class="co">#&gt;   parent.treedata     tidytree</span></span>
<span><span class="co">#&gt;   root.treedata       tidytree</span></span>
<span><span class="co">#&gt;   rootnode.phylo      tidytree</span></span>
<span><span class="co">#&gt;   sibling.phylo       tidytree</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">y</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">y</span>, which will replace the existing scale.</span></span>
<span></span>
<span><span class="va">SingleValueTree</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Tree built with Percent Identity and labelled with Strand Randomisation Index scores"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridis/reference/scale_viridis.html" class="external-link">scale_color_viridis</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Strand Randomisation Index"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtree</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom-hilight.html" class="external-link">geom_hilight</a></span><span class="op">(</span>node <span class="op">=</span> <span class="fl">8</span>, fill <span class="op">=</span> <span class="st">"lightblue1"</span>, alpha <span class="op">=</span> <span class="fl">.2</span>, type <span class="op">=</span> <span class="st">"gradient"</span>, gradient.direction <span class="op">=</span> <span class="st">'tr'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtree</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom-hilight.html" class="external-link">geom_hilight</a></span><span class="op">(</span>node <span class="op">=</span> <span class="fl">9</span>, fill <span class="op">=</span> <span class="st">"pink"</span>, alpha <span class="op">=</span> <span class="fl">.2</span>, type <span class="op">=</span> <span class="st">"gradient"</span>, gradient.direction <span class="op">=</span> <span class="st">'tr'</span><span class="op">)</span></span></code></pre></div>
<p><img src="ScrambledTreeBuilder_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MultiValueTree</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/visualizeTree.html">visualizeTree</a></span><span class="op">(</span><span class="va">tibbleWithMultipleValues</span>, <span class="va">tibbleWithMultipleValues</span><span class="op">$</span><span class="va">Strand_Randomisation_Index</span>, ynudge <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">y</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">y</span>, which will replace the existing scale.</span></span>
<span></span>
<span><span class="va">MultiValueTree</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Tree labeled with Strand Randomisation Index and Percent Identity (built with Percent Identity)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridis/reference/scale_viridis.html" class="external-link">scale_color_viridis</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Strand Randomisation Index"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggnewscale</span><span class="fu">::</span><span class="fu"><a href="https://eliocamp.github.io/ggnewscale/reference/new_scale.html" class="external-link">new_scale_colour</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtree</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_label</a></span><span class="op">(</span><span class="fu">ggtree</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">Percent_Identity</span>, digits <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>, color <span class="op">=</span> <span class="va">Percent_Identity</span><span class="op">)</span>, label.size <span class="op">=</span> <span class="fl">0.25</span>, size <span class="op">=</span> <span class="fl">3</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span>, label.padding <span class="op">=</span> <span class="fu">ggtree</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">0.15</span>, <span class="st">"lines"</span><span class="op">)</span>, nudge_y <span class="op">=</span> <span class="op">-</span><span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridis/reference/scale_viridis.html" class="external-link">scale_color_viridis</a></span><span class="op">(</span>option <span class="op">=</span> <span class="st">"magma"</span>, name <span class="op">=</span> <span class="st">"Percent Identity"</span><span class="op">)</span></span></code></pre></div>
<p><img src="ScrambledTreeBuilder_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Noa Brenner.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
